app/services/auto_assignment/assignment_service.rb

```diff
@@ -0,0 +1,90 @@
+class AutoAssignment::AssignmentService
+  pattr_initialize [:inbox!]
+
+  def perform_bulk_assignment(limit: 100)
+    return 0 unless inbox.enable_auto_assignment
+
+    assigned_count = 0
+
+    unassigned_conversations(limit).each do |conversation|
+      assigned_count += 1 if perform_for_conversation(conversation)
+    end
+
+    assigned_count
+  end
+
+  private
+
+  def perform_for_conversation(conversation)
+    return false unless assignable?(conversation)
+
+    agent = find_available_agent
+    return false unless agent
+
+    assign_conversation(conversation, agent)
+  end
+
+  def assignable?(conversation)
+    conversation.status == 'open' &&
+      conversation.assignee_id.nil?
+  end
+
+  def unassigned_conversations(limit)
+    scope = inbox.conversations.unassigned.open
+
+    scope = if assignment_config['conversation_priority'] == 'longest_waiting'
```

Comment:
Rails already provides a method `earliest_created?` and `longest_waiting?` you could use that instead of a brittle string matching. Similarly for `round_robin?` and `balanced?` 

We also have an option for setting up prefix and suffix for these methods, [ref](https://api.rubyonrails.org/classes/ActiveRecord/Enum.html#:~:text=You%20can%20use%20the%20%3Aprefix%20or%20%3Asuffix%20options%20when%20you%20need%20to%20define%20multiple%20enums%20with%20same%20values.%20If%20the%20passed%20value%20is%20true%2C%20the%20methods%20are%20prefixed/suffixed%20with%20the%20name%20of%20the%20enum.%20It%20is%20also%20possible%20to%20supply%20a%20custom%20value%3A)

We don't need the suffix or prefix for now, but we can use the enum methods anyway